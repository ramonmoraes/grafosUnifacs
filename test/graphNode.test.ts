import GraphNode from "../src/graphNode";

describe("GraphNode", () => {
  test("remove attribute", () => {
    const node = new GraphNode({
      identifier: "foo",
      attributes: {
        key: "val"
      }
    });

    expect("key" in node.attributes).toBeTruthy();
    node.removeAttribute("key");
    expect("key" in node.attributes).toBeFalsy();
  });

  test("snapshot match", () => {
    const node = new GraphNode({
      identifier: "foo",
      attributes: {
        key: "val"
      }
    });
    expect(node).toMatchSnapshot();
  });
});
