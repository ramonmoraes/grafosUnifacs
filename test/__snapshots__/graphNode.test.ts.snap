// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot match 1`] = `
GraphNode {
  "attributes": Object {
    "key": "val",
  },
  "identifier": "foo",
  "removeAttribute": [Function],
}
`;
